test/functional/transactions_tests.js

```diff
@@ -205,6 +205,48 @@ describe('Transactions', function() {
       }
     });
   });
+
+  describe('startTransaction', function() {
+    it('should error if transactions are not supported', {
+      metadata: { requires: { topology: ['sharded'], mongodb: '>4.0.0' } },
+      test: function(done) {
+        if (this.configuration.usingUnifiedTopology()) {
+          return this.skip();
+        }
+
+        const topology = new core.Mongos();
+        const sessionPool = new sessions.ServerSessionPool(topology);
+        const session = new sessions.ClientSession(topology, sessionPool);
+
+        expect(() => session.startTransaction()).to.throw(
+          'Transactions are not supported on sharded clusters in MongoDB < 4.2.'
+        );
+
+        session.endSession(done);
+        sessionPool.endAllPooledSessions();
+      }
+    });
+
+    it('should error if transactions are not supported (unified topology)', {
+      metadata: { requires: { topology: ['sharded'], mongodb: '>4.0.0' } },
+      test: function(done) {
+        if (!this.configuration.usingUnifiedTopology()) {
+          return this.skip();
+        }
+
+        const topology = new core.Topology();
+        const sessionPool = new sessions.ServerSessionPool(topology);
+        const session = new sessions.ClientSession(topology, sessionPool);
+
+        expect(() => session.startTransaction()).to.throw(
+          'Transactions are not supported on sharded clusters in MongoDB < 4.2.'
+        );
+
+        session.endSession(done);
+        sessionPool.endAllPooledSessions();
```

Comment:
this will race with the previous line. You'd need to do:
```
session.endSession(() => {
  sessionPool.endAllPooledSessions(done);
});
```

Reply:
Why will this race? I was following the pattern in the preceding test, `'should provide a useful error if a Promise is not returned'`.

Reply:
There is a chance that the test will end and trigger the session leak checker, because done is called _some time_ before `sessionPool.endAllPooledSessions();`